app/views/pets/show.html.erb

```diff
@@ -1,6 +1,12 @@
 <h2> <%= @pet.name %>'s Information </h2>
 <img src="<%=  @pet.image %>" alt="<%= @pet.image %>" style="width:200px">
-<%=button_to "Favorite this pet.", "/favorites/#{@pet.id}", method: :patch %>
+
+<% if !favorites.has_pet?(@pet.id) %>
+  <%= button_to 'Favorite this pet.', "/favorites/#{@pet.id}", method: :patch, class: "button" %>
+<% else %>
+  <%= button_to 'Remove this pet from favorites.', "/favorites/#{@pet.id}", method: :delete, class: "button" %>
+<% end %>
+
```

Comment:
Again using the has_pet helper method to either show the button to add to favorites, or the button to remove.

Reply:
Clarification for me: what's the use for the bang before favorites?